contracts/VotingToChangeKeys.sol

```diff
@@ -46,6 +46,7 @@ contract VotingToChangeKeys is VotingToChange, EnumKeyTypes {
         address _newPayoutKey
     ) public returns(uint256) {
         IKeysManager keysManager = _getKeysManager();
+        require(!keysManager.hasMiningKeyBeenRemoved(_newMiningKey));
```

Comment:
I think the best place for such checking is inside the `KeysManager.checkIfMiningExisted` function because the ballot can also be created by `VotingToChangeKeys.createBallot`. Besides that, there can be another type of a ballot when we just swap the mining key without its removing/adding.

`KeysManager.checkIfMiningExisted` function is used both for adding mining key and for its swapping (we should restrict swapping some mining key to the key which has already been added before).

`KeysManager.checkIfMiningExisted` function is called by `BallotsStorage._areKeyAddingBallotParamsValid` and `_areKeySwapBallotParamsValid` functions which are in turn called by `BallotsStorage.areKeysBallotParamsValid` which is called by `VotingToChangeKeys.createBallot`. Note that `VotingToChangeKeys.createBallotToAddNewValidator` already calls the `VotingToChangeKeys.createBallot`.

Reply:
Sorry, I was inattentive. I will fix it